### PR TITLE
avoid removal of spaces instead of CONDA_BUILD_SYSROOT

### PR DESCRIPTION
**Pull request by katietz**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Review by xhochy — COMMENTED** [review]
This should probably be backported to older releases, too?

File: recipe/build_base.sh [hunk 1]
```diff
@@ -450,8 +450,11 @@ pushd "${PREFIX}"/lib/python${VER}
   # Append the conda-forge zoneinfo to the end
   sed -i.bak "s@zoneinfo'@zoneinfo:$PREFIX/share/tzinfo'@g" sysconfigfile
   # Remove osx sysroot as it depends on the build machine
-  sed -i.bak "s@-isysroot @@g" sysconfigfile
-  sed -i.bak "s@$CONDA_BUILD_SYSROOT @@g" sysconfigfile
+  # be sure CONDA_BUILD_SYSROOT has value, as other we will remove here instead spaces
+  if [[ ${HOST} =~ .*darwin.* ]] && [[ -n ${CONDA_BUILD_SYSROOT} ]]; then
```
[the remark this answers is not in the document]
Comment (katietz) [reply recipe/build_base.sh:454]: well, I can use.  But other place dealing here with CONDA_BUILD_SYSROOT still uses HOST too. So I just used existing patterns

**Comment by katietz** [comment]
sure, if path has final state, it can be backported.  I just checked that older version in anacondaRecipes weren't affected, and assumed that this line is Python 3.10 specific

**Comment by katietz** [comment]
checking for backport, 3.9 seems to be the only branch affected by the same issue

**Comment by katietz** [comment]
@isuruf could you comment or approve on it?

**Comment by isuruf** [comment]
Thanks

**Pull request merged**